output/write.go

```diff
@@ -0,0 +1,111 @@
+package output
```

Comment:
All of the code in this file is extracted from write.go at the top level: WriteFileAtomically is a new function extracted from OutputDir.Write(Secret), and the rest is straight moved.